qark/modules/findPending.py

```diff
@@ -21,456 +22,499 @@
-					if type(current.type) is m.Type:
-						if hasattr(current.type.name,'value'):
-							if current.type.name.value == "Intent":
-								for v in current.variable_declarators:
-									if str(v.variable.name) == str(intent):
-										found=True
-										for a in v.initializer.arguments:
-											if type(a) is m.ClassLiteral:
-												explicit=True
-												results=[found,explicit]
-												return results
-			elif type(current) is m.VariableDeclarator:
-				if hasattr(c.initializer,'type'):
-					if hasattr(c.initializer.type.name,'value'):
-						if c.initializer.type.name.value == "Intent":
-							if str(c.variable.name) == str(intent):
-								found=True
-								for a in c.initializer.arguments:
-									if type(a) is m.ClassLiteral:
-										explicit=True
-										results=[found,explicit]
-										return results
-								explicit=False
-								results=[found,explicit]
-								return results
-			else:
-				if hasattr(current,'_fields'):
-					for r in current._fields:
-						results=intent_digger(getattr(current,r),intent)
-						if results[0] == True:
-							return results
-							
-	elif type(t) is list:
-		for y in t:
-			results=intent_digger(y,intent)
-			if results[0] == True:
-				return results
-
-	return results
+def start(queue, height):
+    """ Start finding pending intents """
+    results = []
+    global tree
+    global current_file
+    count = 0
+
+    # TODO - Look for use of fillIn method which can make this a much more exploitable condition
+    for j in common.java_files:
+        count = count + 1
+        pub.sendMessage('progress', bar='Pending Intents', percent=round(count * 100 / common.java_files.__len__()))
+        current_file = j
+        try:
+            tree = parser.parse_file(j)
+        except ValueError as e:
+            continue
+        # TODO - Need to add scanning of the imports, to see if Intent or PendingIntent is extended, was working on it,
+        # but the one issue where it arose was non-trivial, so I gave up for now
+
+        try:
+            for import_decl in tree.import_declarations:
+                if 'PendingIntent' in import_decl.name.value:
+                    for type_decl in tree.type_declarations:
+                        if type(type_decl) is m.ClassDeclaration:
+                            for t in type_decl.body:
+                                try:
+                                    empty_intent(t, j, results)
+                                except Exception as e:
+                                    common.logger.debug("Problem in empty intent function of findPending.py: " + str(e))
+                                    common.parsingerrors.add(str(current_file))
+                                for f in t._fields:
+                                    # dynamically parse each token where is f is the field and t is the token
+                                    try:
+                                        recurse(f, t, results)
```

Comment:
It's non-obvious what this function does. Rename it to something like `recursive_search_for_<whatever you're looking for>` at least enough for context